src/qt/bitcoin.cpp

```diff
@@ -489,6 +496,22 @@ int GuiMain(int argc, char* argv[])
     bool prune = false; // Intro dialog prune check box
     // Gracefully exit if the user cancels
     if (!Intro::showIfNeeded(*node, did_show_intro, prune)) return EXIT_SUCCESS;
+    // Now check if the custom config files are accecible
+    // If you run Bitcoin Qt the first time (no datadir) it will simply create the config files at that location
+    if(!qt_settings_path.empty())
+    {
+        QSettings settings;
+        std::string settingspath = settings.fileName().toUtf8().constData();
+        // TODO: After C++17 move, port this to std::filesystem
+        std::fstream f;
+        f.open(settingspath.c_str());
+        // File could not be opened
+        if (f.fail()) {
+            QMessageBox::critical(nullptr, PACKAGE_NAME,
+                QObject::tr("Error: Cannot open config file at \"%1\"").arg(QString::fromStdString(qt_settings_path)));
```

Comment:
IMHO that message could be more concrete - that `config file` is a bit goo vague. Maybe mentioning `GUI settings file` would be more informative (since it's for **-guisettings**dir).